src/main/java/org/spongepowered/api/map/decoration/MapDecoration.java

```diff
@@ -0,0 +1,164 @@
+     */
+    int getY();
+
+    /**
+     * Sets the position of where the MapDecoration is on Maps,
+     * or where it would be if applied to a {@link org.spongepowered.api.map.MapInfo}
+     * 0,0 is the center of the map
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param position Vector2i world x and y cords
+     */
+    void setPosition(Vector2i position);
+
+    /**
+     * Sets the X position of where this MapDecoration is on a map
+     * or where it will be when applied
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param x world cordinate
+     */
+    void setX(int x);
+
+    /**
+     * Sets the X position of where this MapDecoration is on a map.
+     * 0 is the center of the map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param y pos
+     */
+    void setY(int y);
+
+    /**
+     * Sets rotation with a Direction. Must be a cardinal/ordinal/secondary ordinal
+     * @param rot Direction, a, secondary ordinal
+     */
+    void setRotation(Direction rot);
+
+    /**
+     * Gets the direction the Map Decoration is pointing
+     * @return Direction, a cardinal/ordinal/secondary
+     */
+    Direction getRotation();
+
+    interface Builder extends DataBuilder<MapDecoration> {
+        /**
+         * Sets the MapDecorationType
+         * @param type MapDecoration to set
+         * @return this Builder, for chaining
+         */
+        Builder type(MapDecorationType type);
+
+        /**
+         * Sets the X position of the MapDecoration
+         * (for when it is applied to a {@link org.spongepowered.api.map.MapInfo})
+         * @param x map coordinate where 0 is the centre of the map.
+         * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+         * @return this Builder, for chaining
+         */
+        Builder x(int x) throws IllegalArgumentException;
+
+        /**
+         * Sets the Y position of the MapDecoration
+         *
+         * (for when it is applied to a {@link org.spongepowered.api.map.MapInfo})
+         * @param y map coordinate where 0 is the centre of the map.
+         * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+         * @return this Builder, for chaining
+         */
+        Builder y(int y) throws IllegalArgumentException;
+
+        /**
+         * Sets the rotation in degrees for this builder
+         * (for when it is applied to a {@link org.spongepowered.api.map.MapInfo})
+         * @param rot Direction, any cardinal/ordinal/secondary ordinal direction. (Not up/down)
+         * @return this Builder, for chaining
+         */
+        Builder rotation(Direction rot) throws IllegalArgumentException;
```

Comment:
Would using the `Rotation` class be better here? That has simple directions in a non-cardinal form and given it's said a map doesn't represent the world, N/S/E/W don't really make as much sense as `TOP`, `BOTTOM` etc.

(Also, JDs say "rotation in degrees" - that's not what's happening here.)

Reply:
I am all for this, but we need 16 directions, whereas currently Rotations currently has 8. Should I add fields for TOP_TOP_LEFT, in the same style as Direction, or just keep using Direction? Have you got better names for the fields?

Reply:
Or just make a `MapDecorationOrientation` psuedo enum and use that.